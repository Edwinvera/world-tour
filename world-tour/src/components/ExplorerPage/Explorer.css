.Africa{
  background-image: url("../../images/ExplorerRegions/Africa.jpg");
  background-repeat: no-repeat;
  background-size: cover;
}

.Europe{
  background-image: url("../../images/ExplorerRegions/Europe.jpg");
  background-repeat: no-repeat;
  background-size: cover;
}

.Americas{
  background-image: url("../../images/ExplorerRegions/Americas.jpg");
  background-repeat: no-repeat;
  background-size: cover;
}

.Asia{
  background-image: url("../../images/ExplorerRegions/Asia.jpg");
  background-repeat: no-repeat;
  background-size: cover;
}
.Oceania{
  background-image: url("../../images/ExplorerRegions/Oceania.jpg");
  background-repeat: no-repeat;
  background-size: cover; 
}
.Polar{
  background-image: url("../../images/ExplorerRegions/Polar.jpg");
  background-repeat: no-repeat;
  background-size: cover;
}
