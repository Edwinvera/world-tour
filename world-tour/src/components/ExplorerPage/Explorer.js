import React, {useState} from 'react'
import { Link } from 'react-router-dom'
import './Explorer.css'
import styled from 'styled-components'

const Main = styled.main`
  justify-content: center;
  text-align: center;
  margin: 225px 0
`

const SectionRandom = styled.section`
  display: flex;
  flex-direction: column;
  align-items: center;
`

const RandoBtn = styled.button`
  margin: 15px 0px;
  height: 35px;
  background-color: rgb(17,124,158);
  color: white;
  border-radius: 5px;
  font-size: 18px;
  cursor: pointer
`
const RandoBox = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  height: 50px;
  width: 300px;
  border: blue dotted 2px;
  background-color: rgba(17,124,158,.2)
`
const RandoCountry = styled.h5`
  font-family: 'Montserrat', sans-serif;
  font-size: 18px
`
const CountryLink = styled(Link)`
  text-decoration: none;
  color: black
`
const RegionLink = styled(CountryLink)`
  text-decoration: none;
  color: black
`
const SectionRegion = styled(SectionRandom)`
  margin: 50px 0
`
const RegionContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 10px;
  height: 300px;
  width: 1000px;
`
const Region = styled.h2`
  font-family: 'Nunito Sans', sans-serif;
  font-size: 36px;
  letter-spacing: 5px;
  padding: 0px 5px;
  color: white;
  background-color: rgba(20,20,20,.1);
  box-shadow: 5px 10px 20px rgb(20,20,20);
`


function Explorer({ uniqueRegions, countries }) {
  
  const [randoCountry, updateCountry] = useState('--?--')

  function handleRandomBtn(e) {
    const countryCount = countries.length
    const randomNumber = Math.floor(Math.random() * countryCount)
    const randomCountry = countries.filter((country,i) => countries.indexOf(country) === randomNumber)

    updateCountry(randomCountry)
    // console.log(randomCountry)
  }

  return (
    <div>
      <Main>
        <SectionRandom>

          <RandoBtn onClick= {(e) => handleRandomBtn()}>Visit a Random Country</RandoBtn>
          
          <RandoBox>
            <CountryLink to='/country/:alpha3code'>
              <RandoCountry>{randoCountry}</RandoCountry>
            </CountryLink>
          </RandoBox>
          
        </SectionRandom>
        <SectionRegion>
          {/* A map loop for regions here. Include a className so I can style the region names with a background image*/}
          {uniqueRegions.map((region, id) => 
            <RegionContainer className={region}>
              <RegionLink to='/region/:alphacode3' key={id}>
                <Region>{region}</Region>
              </RegionLink>
            </RegionContainer>
          )} 
        </SectionRegion>  
      </Main>
      
      
    </div>
  )
}

export default Explorer
