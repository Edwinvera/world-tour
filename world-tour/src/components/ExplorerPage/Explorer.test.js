import renderer from 'react-test-renderer'
import 'jest-styled-components'
import React from 'react'
import Explorer, {RegionContainer, RandoBox, RandoBtn, RandoCountry} from './Explorer'

test('Explorer_RegionContainer component renders', () => {
  const RegionCointainerComponent = renderer.create(<RegionContainer/>).toJSON()
  expect(RegionCointainerComponent).toMatchSnapshot()
})

test('Explorer_RandoBox component renders', () => {
  const RandoBoxComponent = renderer.create(<RandoBox/>).toJSON()
  expect(RandoBoxComponent).toMatchSnapshot()
})

test('Explorer_RandoBtn component renders', () => {
  const RandoBtnComponent = renderer.create(<RandoBtn/>).toJSON()
  expect(RandoBtnComponent).toMatchSnapshot()
})

test('Explorer_RandoCountry component renders', () => {
  const RandoCountryComponent = renderer.create(<RandoCountry/>).toJSON()
  expect(RandoCountryComponent).toMatchSnapshot()
})