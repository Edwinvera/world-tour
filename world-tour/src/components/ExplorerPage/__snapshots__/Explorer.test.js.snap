// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Explorer_RandoBox component renders 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 50px;
  width: 300px;
  border: blue dotted 2px;
  background-color: rgba(17,124,158,.2);
}

<div
  className="c0"
/>
`;

exports[`Explorer_RandoBtn component renders 1`] = `
.c0 {
  margin: 15px 0px;
  height: 35px;
  background-color: rgb(17,124,158);
  color: white;
  border-radius: 5px;
  font-size: 18px;
  cursor: pointer;
}

<button
  className="c0"
/>
`;

exports[`Explorer_RandoCountry component renders 1`] = `
.c0 {
  font-family: 'Montserrat',sans-serif;
  font-size: 18px;
}

<h5
  className="c0"
/>
`;

exports[`Explorer_RegionContainer component renders 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 10px;
  height: 300px;
  width: 1000px;
}

@media (max-width:1000px) {
  .c0 {
    height: 300px;
    width: 860px;
  }
}

@media (max-width:861px) {
  .c0 {
    height: 300px;
    width: 600px;
    margin: 20px;
    border-radius: 10px;
  }
}

@media (max-width:697px) {
  .c0 {
    height: 300px;
    width: 300px;
    margin: 20px;
    border-radius: 10px;
  }
}

<div
  className="c0"
/>
`;
